main.go

```diff
@@ -187,25 +187,44 @@ func useConfig(cmd *cobra.Command) (err error) {
 	}
 
 	// Create cache
-	if viper.GetString("redis.url") != "" {
-		redisCacheVersionsLatest := redis.NewClient(&redis.Options{
-			Addr:     viper.GetString("redis.url"),
-			Password: viper.GetString("redis.password"),
-			DB:       0,
-		})
-		redisCacheVersionsList := redis.NewClient(&redis.Options{
-			Addr:     viper.GetString("redis.url"),
-			Password: viper.GetString("redis.password"),
-			DB:       1,
-		})
+	if redisURL := viper.GetString("redis.addrs"); redisURL != "" {
+		redisOpts := &redis.UniversalOptions{
+			// Either a single address or a seed list of host:port addresses
+			// of cluster/sentinel nodes.
+			Addrs: viper.GetStringSlice("redis.addrs"),
+
+			// The sentinel master name.
+			// Only failover clients.
+			MasterName: viper.GetString("redis.master"),
+
+			// Enables read only queries on slave nodes.
+			ReadOnly: viper.GetBool("redis.read_only_slave"),
+
+			MaxRetries:         viper.GetInt("redis.max_retries"),
+			Password:           viper.GetString("redis.password"),
+			DialTimeout:        viper.GetDuration("redis.dial_timeout"),
+			ReadTimeout:        viper.GetDuration("redis.read_timeout"),
+			WriteTimeout:       viper.GetDuration("redis.write_timeout"),
+			PoolSize:           viper.GetInt("redis.pool_size"),
+			PoolTimeout:        viper.GetDuration("redis.pool_timeout"),
+			IdleTimeout:        viper.GetDuration("redis.idle_timeout"),
+			IdleCheckFrequency: viper.GetDuration("redis.idle_check_frequency"),
+		}
+		optsList := redisOpts
+		optsLatest := redisOpts
+
+		optsList.DB = viper.GetInt("redis.databases.versionsList")
+		optsLatest.DB = viper.GetInt("redis.databases.versionsLatest")
```

Comment:
You are sure that the `optsList.DB` will be taken in account? It looks like `optsList` and `optsLatest` are pointers to the same struct, and changes for one impacts the other.

Reply:
Nice pick, I read the doc too fast. Struct copy like this works only with primitive types